python/tvm/autotvm/database.py

```diff
@@ -142,29 +145,33 @@ def save(self, inp, res, extend=False):
 
     def filter(self, func):
         """
-        Dump all of the records for a particular target
+        Dump all of the records that match the given rule
 
         Parameters
         ----------
         func: callable
-            The signature of the function is bool (MeasureInput, Array of MeasureResult)
+            The signature of the function is (MeasureInput, [MeasureResult]) -> bool
 
         Returns
         -------
-        list of records (inp, result) matching the target
+        list of records in tuple (MeasureInput, MeasureResult) matching the rule
 
         Examples
         --------
         get records for a target
         >>> db.filter(lambda inp, resulst: "cuda" in inp.target.keys)
+        get records with errors
+        >>> db.filter(lambda inp, results: any(r.error_no != 0 for r in results))
         """
         matched_records = list()
         # may consider filtering in iterator in the future
-        for key in self.db:
+        for key in self.db.keys():
```

Comment:
```suggestion
        for key in self.db:
```

should not need `keys`